src/domain/cpu/workers/hello.worker.ts

```diff
@@ -0,0 +1,63 @@
+export interface DataInMessagePayload {
+  offset: number;
+  chunkLength: number;
+  pattern: Uint8Array;
+  text: Uint8Array;
+}
+
+export interface DataOutMessagePayload {
+  matches: Array<number>;
+}
+
+class SolverCPU {
+  private readonly ctx: Worker = (self as unknown) as Worker;
+
+  constructor() {
+    this.ctx.onmessage = this.handleMessage.bind(this);
+  }
+
+  handleMessage(message: MessageEvent<DataInMessagePayload>) {
+    if (message.data.pattern.length < 1 || message.data.text.length < 1) {
+      throw Error("Pattern and text must be at least 1 character long.");
+    }
+    if (message.data.pattern.length > message.data.text.length) {
+      throw Error("Pattern can't be longer than text.");
+    }
+    const matches = this.solve(message.data);
+    this.ctx.postMessage({
```

Comment:
Is it correct with the interface DataOutMessagePayload?

Reply:
Yes

Reply:
I added the expected type of `const matches` to `Array<number>` for clarity.